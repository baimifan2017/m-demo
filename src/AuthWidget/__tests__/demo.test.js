import React from 'react';
import { render, mount } from 'enzyme';
import { Input } from 'antd';
import toJson from 'enzyme-to-json';
import { AuthWidget } from 'seid';

describe('AuthWidget', () => {
  it('No Auth', () => {
    const AuthInput = AuthWidget(Input);
    const inputRender = mount(<AuthInput authCode="authCode" />);
    expect(inputRender.contains(<Input />)).toEqual(false);
    expect(toJson(inputRender)).toMatchSnapshot();
  });

  it('has Auth', () => {
    const AuthInput = AuthWidget(Input);
    const authInput = mount(<AuthInput authCode="authCode" authorities={['authCode']} />);
    expect(authInput.contains(<Input />)).toEqual(true);
  });

  it('no Contorl', () => {
    const AuthInput = AuthWidget(Input);
    const inputRender = render(<AuthInput />);
    expect(toJson(inputRender)).toMatchSnapshot();
  });
});
