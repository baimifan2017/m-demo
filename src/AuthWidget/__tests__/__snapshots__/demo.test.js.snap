// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthWidget No Auth 1`] = `
<Widget
  authCode="authCode"
/>
`;

exports[`AuthWidget no Contorl 1`] = `
<input
  class="ant-input"
  type="text"
  value=""
/>
`;
